File: Perl/counter.pl
#/usr/bin/perl -w

###############################
#                             #
# FILE      : counter.pl      #
# AUTHOR    : Fernando Freire #
# DATE      : 04/03/2013      #
#                             #
###############################

use strict;

print "Enter a count: ";
# In compliance with Part One's requirements we will explicity read from
# standard input.
chomp(my $count = <STDIN>);

my @values;
print "Enter a value: ";
while(<>) {
    # We use `unless` here to ensure that we are not printing our prompt
    # too many times.
    print "Enter a value: " unless (@values+1) == $count;
    push(@values,$_);
    last if @values == $count;
}

chomp @values;

my $output = join(', ',@values);

# In compliance with Part One's requirements we will explicity write to
# standard output
print( <STDOUT> , "You entered: $output\n" );

File: Ruby/counter.rb
#!/usr/bin/env ruby

###############################
#                             #
# FILE      : counter.rb      #
# AUTHOR    : Fernando Freire #
# DATE      : 04/16/2013      #
#                             #
###############################

print "Enter a count: "
# Get from STDIN, remove the newline, and convert it to an int
count = gets.chomp.to_i

values = []

# Iterate over the values based on the given count
count.times do
    print "Enter a value: "
    values << gets.chomp
end

puts "You entered: #{values.map{ |value| value }.join(', ')}"

